src/main/scala/com/microsoft/partnercatalyst/fortis/spark/sources/streamfactories/StreamFactoryBase.scala

```diff
@@ -0,0 +1,29 @@
+package com.microsoft.partnercatalyst.fortis.spark.sources.streamfactories
+
+import com.microsoft.partnercatalyst.fortis.spark.logging.FortisTelemetry
+import com.microsoft.partnercatalyst.fortis.spark.sources.streamprovider.{ConnectorConfig, StreamFactory}
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.DStream
+
+import scala.reflect.ClassTag
+
+abstract class StreamFactoryBase[A: ClassTag] extends StreamFactory[A]{
+  override def createStream(ssc: StreamingContext): PartialFunction[ConnectorConfig, DStream[A]] = {
+    case config if canHandle(config) =>
+      val stream = buildStream(ssc, config)
+
+      stream.transform(rdd => {
+        rdd.cache()
+        val batchSize = rdd.count().toInt
+        val streamId = config.parameters.get("streamId").get.toString
+        val connectorName = config.name
+        val telemetry = FortisTelemetry.get()
+        telemetry.logIncomingEventBatch(streamId, connectorName, batchSize)
```

Comment:
Done in [18676bf](https://github.com/CatalystCode/project-fortis-spark/pull/68/commits/18676bf45d160e4d2adf3f333053d803aa63bf3d)